src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -40,19 +40,67 @@
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             return Optional.ofNullable(session.get(CinemaHall.class, id));
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get a cinema hall by id: " + id, e);
+            throw new DataProcessingException("Can't get a cinemaHall by id: " + id, e);
         }
     }
 
     @Override
     public List<CinemaHall> getAll() {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            CriteriaQuery<CinemaHall> criteriaQuery = session.getCriteriaBuilder()
-                    .createQuery(CinemaHall.class);
-            criteriaQuery.from(CinemaHall.class);
-            return session.createQuery(criteriaQuery).getResultList();
+            return session.createQuery("from CinemaHall").list();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get cinemaHalls", e);
+        }
+    }
+
+    @Override
+    public boolean update(CinemaHall cinemaHall) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.update(cinemaHall);
+            transaction.commit();
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get all cinema halls", e);
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("update " + cinemaHall + " failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+
+        return get(cinemaHall.getId()).get().equals(cinemaHall);
+    }
+
+    @Override
+    public boolean delete(Long id) {
```

Comment:
Consider changing the return type of the `delete` method to `void`. The current approach of returning a boolean by checking if the entity still exists is not reliable for determining if the delete was successful.